src/main/java/org/folio/rest/impl/TenantReferenceAPI.java

```diff
@@ -0,0 +1,291 @@
+   *
+   * @param headers
+   * @param endPoint
+   * @param handler
+   */
+  private void loadSampleDataForEndpoint(Map<String, String> headers, String endPoint, Handler<AsyncResult<Void>> handler) {
+    log.info("load Sample data for: " + endPoint + " begin");
+    String okapiUrl = headers.get("X-Okapi-Url-to");
+    if (okapiUrl == null) {
+      log.warn("No X-Okapi-Url-to. Headers: " + headers);
+      handler.handle(Future.failedFuture("No X-Okapi-Url-to header"));
+      return;
+    }
+    log.info("load Sample Data....................");
+    List<String> jsonList = new LinkedList<>();
+    try {
+      List<InputStream> streams = getStreamsfromClassPathDir(RESOURCES_PATH + endPoint);
+      for (InputStream stream : streams) {
+        jsonList.add(IOUtils.toString(stream, "UTF-8"));
+      }
+    } catch (URISyntaxException ex) {
+      handler.handle(Future.failedFuture("URISyntaxException for path " + endPoint + " ex=" + ex.getLocalizedMessage()));
+      return;
+
+    } catch (IOException ex) {
+      handler.handle(Future.failedFuture("IOException for path " + endPoint + " ex=" + ex.getLocalizedMessage()));
+      return;
+    }
+    final String endPointUrl = okapiUrl + ORDERS_STORAGE_PREFIX_URL + endPoint;
+    List<Future> futures = new LinkedList<>();
+    for (String json : jsonList) {
+      Future<Void> future = Future.future();
+      futures.add(future);
+      if(isUpdateModule)
+        putData(headers, endPointUrl, json, future);
+      else
+        postData(headers, endPointUrl, json, future);
+    }
+    CompositeFuture.all(futures).setHandler(asyncResult -> {
+      log.info("Sample Data load {} done. success={}", endPoint, asyncResult.succeeded());
+      if (asyncResult.failed()) {
+        handler.handle(Future.failedFuture(asyncResult.cause().getLocalizedMessage()));
+      } else {
+        handler.handle(Future.succeededFuture());
+      }
+    });
+  }
+
+  private void postData(Map<String, String> headers, final String endPointUrl, String json, Future<Void> future) {
+    HttpClientRequest req = httpClient.postAbs(endPointUrl, responseHandler -> {
+      if (responseHandler.statusCode() >= 200 && responseHandler.statusCode() <= 299) {
+        future.handle(Future.succeededFuture());
+      } else {
+        future.handle(Future.failedFuture("POST " + endPointUrl + " returned status " + responseHandler.statusCode()));
+      }
+    });
+    for (Map.Entry<String, String> headerEntry : headers.entrySet()) {
+      String header = headerEntry.getKey();
+      if (header.startsWith("X-") || header.startsWith("x-")) {
+        req.headers().add(header, headerEntry.getValue());
+      }
+    }
+    req.headers().add("Content-Type", "application/json");
+    req.headers().add("Accept", "application/json, text/plain");
+    req.end(json);
+  }
+
+
+  private void putData(Map<String, String> headers, final String endPointUrl, String json, Future<Void> future) {
```

Comment:
Not a huge deal but it would be nice to abstract out the duplicate code between putData and postData

Reply:
Sure